javaparser-core/src/main/java/com/github/javaparser/printer/configuration/PrinterConfiguration.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2011, 2013-2020 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.printer.configuration;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import com.github.javaparser.printer.Printable;
+import com.github.javaparser.printer.configuration.Indentation.IndentType;
+import com.github.javaparser.utils.Utils;
+
+/**
+ * Configuration options for the {@link Printable}.
+ */
+public class PrinterConfiguration implements ConfigurablePrinter {
+    
+    public enum ConfigOption {
+        ORDER_IMPORTS(Boolean.class), 
+        PRINT_COMMENTS(Boolean.class), 
+        PRINT_JAVADOC(Boolean.class), 
+        SPACE_AROUND_OPERATORS(Boolean.class), 
+        COLUMN_ALIGN_PARAMETERS(Boolean.class), 
+        COLUMN_ALIGN_FIRST_METHOD_CHAIN(Boolean.class),
+        /**
+         * Indent the case when it is true, don't if false
+         * switch(x) {            switch(x) {
+         *    case 1:             case 1:
+         *        return y;           return y;
+         *    case 2:             case 2:
+         *        return z;           return x;
+         *}                       }
+         */
+        INDENT_CASE_IN_SWITCH(Boolean.class),
+        /**
+         * By default enum constants get aligned like this:
+         * <pre>
+         *     enum X {
+         *        A, B, C, D
+         *     }
+         * </pre>
+         * until the amount of constants passes this value (5 by default).
+         * Then they get aligned like this:
+         * <pre>
+         *     enum X {
+         *        A,
+         *        B,
+         *        C,
+         *        D,
+         *        E,
+         *        F,
+         *        G
+         *     }
+         * </pre>
+         * Set it to a large number to always align horizontally.
```

Comment:
Suggestion:
`Set it to a very large number (e.g. {@code Integer.MAX_VALUE} to always align horizontally.`

Reply:
It's from the legacy code.

Reply:
In fact i think that these options are related to a specific visitor implementation. So i think there is another refactoring decision to take to isolate the option interface and the option implementation. But I would do this soon in another PR.